pkg/ingester/ingester_v2.go

```diff
@@ -233,6 +238,13 @@ func (i *Ingester) startingV2(ctx context.Context) error {
 		return errors.Wrap(err, "opening existing TSDBs")
 	}
 
+	// Scan and open backfill TSDB's that already exist on disk.
+	// TODO(codesome): run this in parallel with opening main TSDBs
```

Comment:
Is this extra complexity worth? I'm not sure. Having backfilling TSDBs shouldn't be the normal case and I'm not sure adding extra complexity here is worth. I feel fine if we (concurrently) open all existing TSDBs first, and then we (concurrently) open all backfill TSDBs.

Reply:
Yeah that sounds fine, we don't need additional complexity for this.